core/src/main/java/io/grpc/internal/GrpcUtil.java

```diff
@@ -163,31 +163,38 @@
   public static final long DEFAULT_KEEPALIVE_TIMEOUT_NANOS = TimeUnit.MINUTES.toNanos(2);
 
   /**
-   * Maps HTTP error response status codes to transport codes.
+   * Maps HTTP error response status codes to transport codes, as defined in <a
+   * href="https://github.com/grpc/grpc/blob/master/doc/http-grpc-status-mapping.md">
+   * http-grpc-status-mapping.md</a>. Never returns a status for which {@code status.isOk()} is
+   * {@code true}.
    */
   public static Status httpStatusToGrpcStatus(int httpStatusCode) {
-    // Specific HTTP code handling.
+    return httpStatusToGrpcCode(httpStatusCode).toStatus()
+        .withDescription("HTTP status code " + httpStatusCode);
+  }
+
+  private static Status.Code httpStatusToGrpcCode(int httpStatusCode) {
+    if (httpStatusCode > 100 && httpStatusCode < 200) {
+      // 1xx. These headers should have been ignored.
+      return Status.Code.INTERNAL;
```

Comment:
The doc already calls out what should happen with 100: "Technically, 1xx should have the entire header skipped and a subsequent header be read. See RFC 7540 §8.1."